src/main/java/org/opennars/control/concept/ProcessJudgment.java

```diff
@@ -167,10 +169,14 @@ protected static boolean isExecutableHypothesis(Task task, final DerivationConte
     protected static void addToTargetConceptsPreconditions(final Task task, final DerivationContext nal, final Concept alternativeTarget) {
         Set<Term> targets = new HashSet<Term>();
         if(alternativeTarget == null) { 
-            //add to all components
-            Map<Term, Integer> ret = ((Implication)task.getTerm()).getPredicate().countTermRecursively(null);
-            for(Term r : ret.keySet()) {
-                targets.add(r);
+            //add to all components, unless it is an negation
+            if(((Implication)task.getTerm()).getPredicate() instanceof Negation) {

```

Comment:
TODO change: No it's not about whether it is a Negation, but whether it contains variables, as there is no point in letting a precondition compete for existence in a component concept if it can't match with any other than goals with itself's predicate.